src/Tools/dotnet-monitor/ParameterCapturing/CaptureParametersOperation.cs

```diff
@@ -114,6 +141,7 @@ public async Task ExecuteAsync(CancellationToken token)
                 settings.OnCapturingFailed += OnCapturingFailed;
                 settings.OnServiceStateUpdate += OnServiceStateUpdate;
                 settings.OnUnknownRequestId += OnUnknownRequestId;
+                settings.OnParametersCaptured += OnParametersCaptured;
```

Comment:
We shouldn't use a store here to buffer all the parameters as it will:
- Lead to unbounded memory growth for infinite duraiton sessions
- Not write out data as its available like all of the other artifacts that are on-demand

Ideally when we receive a new set of parameters we immediately format them and write them to the output stream, letting the caller deal with memory restrictions, eviction policies, etc. 

I'm assuming this is modeled after how exceptions does this. Exceptions (and metrics) is a bit different from the other artifacts as it is continuously collecting data unprompted and also needs to do correlations between the pieces of data as new information is learned (e.g. inner exceptions).

Parameters are on-demand (and now) just like logs and traces, so the user has to request them and has to actively listen for the results, or results are getting sent to an egress provider. Because of this there's no need to cache data on dotnet-monitors side.

Reply:
I refactored the code to use a `CapturedParametersWriter` instead of the store. The writer queues the captured methods on a separate channel when the `OnParametersCaptured` event gets triggered and on a separate loop the data is written to the stream through the abstract `CapturedParametersFormatter` (has two implementations: `CapturedParametersTextFormatter` and `CapturedParametersJsonFormatter`)
I also introduced the `NewlineDelimitedJson` result format since it might be easier to parse if we expect data to come out in chunks.

Reply:
> I also introduced the NewlineDelimitedJson result format since it might be easier to parse if we expect data to come out in chunks.

We should mirror how `/logs` work here for consistency instead of rolling something different. `/logs` produces json sequence (or ndjson), see this file for how it does the formatting:

https://github.com/dotnet/dotnet-monitor/blob/8b4c2545bdd5abdfdc16a42abbfdb0c32554ed4e/src/Microsoft.Diagnostics.Monitoring.WebApi/StreamingLogger.cs#L98